src/listeners/__tests__/browser.spec.js

```diff
@@ -0,0 +1,46 @@
+import tape from 'tape-catch';
+import sinon from 'sinon';
+import Context from '../../utils/context';
+import StorageFactory from '../../storage';
+import SettingsFactory from '../../utils/settings';
+import BrowserSignalListener from '../browser';
+
+const UNLOAD_DOM_EVENT = 'unload';
+
+const windowAddEventListenerSpy = sinon.spy();
+const windowRemoveEventListenerSpy = sinon.spy();
+
+sinon.stub(window, 'addEventListener').callsFake(windowAddEventListenerSpy);
```

Comment:
You could have only used a spy (or a stub if you have to prevent the real method being executed)